src/DataValues/LanguageCodeValue.php

```diff
@@ -49,15 +64,22 @@ protected function parseUserValue( $userValue ) {
 
 		// Checks whether the language tag is valid in MediaWiki for when
 		// it is not executed in a query context
-		if ( !$this->getOption( self::OPT_QUERY_CONTEXT ) && !Localizer::isKnownLanguageTag( $languageCode ) ) {
+		
+		// ensure non-standard language codes are mapped to
+		// their canonical form (e.g. de-x-formal to de-formal)
+
+		$mappedLanguageCode = !in_array( $languageCode, $this->nonstandardLanguageCodeMapping ) ? $languageCode
```

Comment:
`in_array` isn't needed. `array_search( $languageCode, $this->nonstandardLanguageCodeMapping ) ?: $languageCode` does the same thing.

Reply:
Yes bcp47 should be fine to use when formatting language codes for display. The list of deprecated language codes is changing over time, so it's possible the behavior is not the same across all MediaWiki versions.